src/routes/chat/components/messages/Messages.js

```diff
@@ -1,41 +1,58 @@
-import React from 'react';
+import React, { useRef, useEffect } from 'react';
 import map from 'lodash/map';
 import PropTypes from 'prop-types';
 import isEmpty from 'lodash/isEmpty';
 
 import Message from '../message';
 
-const testValueIsMe = true;
+const testUserId = 1;
 
-function Messages({ items }) {
+function Messages({ changeInputToEdit, chatRemoveMessage, items }) {
   console.info('In messages', items);
+  const messageReference = useRef(null);
+  useEffect(() => {
+    messageReference.current.scrollIntoView(0, items[items.length - 1]);
+  }, [items]);
   return (
-    <div>
-      {!isEmpty(items) &&
+    <div ref={messageReference}>
+      {!isEmpty(items) ? (
         map(items, item => {
+          // For new message
+          const createdAt = !item.createdAt ? `${new Date()}` : item.createdAt;
+          const id = !item.id ? Math.random() : item.id;
```

Comment:
Do you realise that `Math.random()` will return new value on each render? Better assign some pseudoId in the code which creates an item

Reply:
May I use Math.random() outside the function Messages (const temporaryId = Math.random();), and if item have not "id" I use "temporaryId" ( key={ item.id || temporaryId } ) ?